CHANGELOG.md

```diff
@@ -1,3 +1,29 @@
+CHANGES IN VERSION 1.10.0 (from 1.9.0)
+===================================
+
+Appium 1.10.0 is a minor release.
+
+#### General
+* Bring Espresso Driver out of beta and into general availability
+* Discontinue support for Xcode 8
+
+#### iOS
+* Add biometrics (touchId, faceId) [methods](http://appium.io/docs/en/commands/mobile-command/#ios-xcuitest-only) that can enroll/unenroll biometrics and send matching/non-matching biometric inputs
+
+#### Android
+* Add [mobile:](http://appium.io/docs/en/commands/mobile-command) command for performing editor actions [#428](https://github.com/appium/appium-android-driver/pull/428)
+* Send custom keystore [capabilities] to ADB (useKeystore, keystorePath, keystorePassword, keyAlias, keyPassword) [#431](https://github.com/appium/appium-android-driver/pull/431)
+* Deprecate `unlock helper` APK [#435](https://github.com/appium/appium-android-driver/pull/435)
+* Add `localeScript` capability to set script in `locale` (https://developer.android.com/reference/java/util/Locale) [#460](https://github.com/appium/appium-android-driver/pull/460)
+* Fix [bug](https://github.com/appium/appium/issues/11619) calling `getRect` in MJSONWP mode [#240](https://github.com/appium/appium-uiautomator2-driver/pull/240)
+* Fix pin entry error on Samsung devices that wasn't correctly locating the "Enter" button [#458](https://github.com/appium/appium-android-driver/pull/458)
+
+#### Android (UiAutomator2)
+* Add [mobile:](http://appium.io/docs/en/commands/mobile-command) command for retrieving device information [#221](https://github.com/appium/appium-uiautomator2-driver/pull/211)
+* Add device information in returned capabilities [#214](https://github.com/appium/appium-uiautomator2-driver/pull/214)
+* Add support for `.apks` bundles [#233](https://github.com/appium/appium-uiautomator2-driver/pull/233)
+* Fix bug with W3C timeouts call [#239](https://github.com/appium/appium-uiautomator2-driver/pull/239)
+
```

Comment:
There was a major refactoring done for UIA2 regarding XML source generation and xpath search

Reply:
Another general Android perf improvement is related to permission setting and some package-related operations (like isAppInstalled)

Reply:
+ Refactored and fixed W3C Actions support

Reply:
Should I write it like

```
* Fixed a bug with XML source generation that was causing xpath search failures
* Made improvements to performance by modifying permission settings

Reply:
Regarding the first one I'd rather write `Refactored XML source generation and xpath search in order to fix known bugs and to improve the general performance of these operations`